DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -256,12 +257,12 @@ internal DMVisitorExpression(DMObject dmObject, DMProc proc, DreamPath? inferred
 
         public void VisitLogicalAndAssign(DMASTLogicalAndAssign land) {
             var lhs = DMExpression.Create(_dmObject, _proc, land.A, _inferredPath);
-            var rhs = DMExpression.Create(_dmObject, _proc, land.B, _inferredPath);
+            var rhs = DMExpression.Create(_dmObject, _proc, land.B, lhs.Path ?? _inferredPath);
```

Comment:
This should just be `lhs.Path`, similar to `VisitAssign()`

`var1 |= var2 |= new()` does not use `var1`'s type if `var2` doesn't have one.
```suggestion
            var rhs = DMExpression.Create(_dmObject, _proc, land.B, lhs.Path);
```